iothub_client/inc/iothub_client.h

```diff
@@ -37,354 +45,109 @@ extern "C"
 #endif
 
     /**
-    * @brief    Creates a IoT Hub client for communication with an existing
-    *             IoT Hub using the specified connection string parameter.
-    *
-    * @param    connectionString    Pointer to a character string
-    * @param    protocol            Function pointer for protocol implementation
-    *
-    *            Sample connection string:
-    *                <blockquote>
-    *                    <pre>HostName=[IoT Hub name goes here].[IoT Hub suffix goes here, e.g., private.azure-devices-int.net];DeviceId=[Device ID goes here];SharedAccessKey=[Device key goes here];</pre>
-    *                   <pre>HostName=[IoT Hub name goes here].[IoT Hub suffix goes here, e.g., private.azure-devices-int.net];DeviceId=[Device ID goes here];SharedAccessSignature=SharedAccessSignature sr=[IoT Hub name goes here].[IoT Hub suffix goes here, e.g., private.azure-devices-int.net]/devices/[Device ID goes here]&sig=[SAS Token goes here]&se=[Expiry Time goes here];</pre>
-    *                </blockquote>
-    *
-    * @return    A non-NULL @c IOTHUB_CLIENT_HANDLE value that is used when
-    *             invoking other functions for IoT Hub client and @c NULL on failure.
+    * @warning IoTHubClient_CreateFromConnectionString is deprecated.  Use IoTHubDeviceClient_CreateFromConnectionString instead.
     */
     MOCKABLE_FUNCTION(, IOTHUB_CLIENT_HANDLE, IoTHubClient_CreateFromConnectionString, const char*, connectionString, IOTHUB_CLIENT_TRANSPORT_PROVIDER, protocol);
 
     /**
-    * @brief    Creates a IoT Hub client for communication with an existing IoT
-    *             Hub using the specified parameters.
-    *
-    * @param    config    Pointer to an @c IOTHUB_CLIENT_CONFIG structure
-    *
-    *            The API does not allow sharing of a connection across multiple
-    *            devices. This is a blocking call.
-    *
-    * @return    A non-NULL @c IOTHUB_CLIENT_HANDLE value that is used when
-    *             invoking other functions for IoT Hub client and @c NULL on failure.
+    * @warning IoTHubClient_Create is deprecated.  Use IoTHubDeviceClient_Create instead.
```

Comment:
Would we want something like this?

```c
#ifdef _MSC_VER
#define DEPRECATED __declspec(deprecated)
#elif defined(__GNUC__) | defined(__clang__)
#define DEPRECATED __attribute__((__deprecated__))
#else
#define DEPRECATED
#endif

//usage:
DEPRECATED void foo(int bar);
```

giving users a compile time warning? Or would we also get these if these are used underneath our APIs

Reply:
I'd love something like that, obviously with a namespace fixup so we don't camp on `DEPRECATED`.

But the `MOCKABLE_FUNCTION` get in the way and I don't want to try and shoe-horn anything else around that even were in possible in theory.  I've put a comment in #1342 draft PR that if that ever comes in, we should reconsider this issue.